src/nnet/nnet-parametric-relu.h

```diff
@@ -0,0 +1,266 @@
+    int32 alpha_num_elem = alpha_.Dim();
+    int32 beta_num_elem = beta_.Dim();
+    gradient->Range(0, alpha_num_elem).CopyFromVec(Vector<BaseFloat>(alpha_corr_));
+    gradient->Range(alpha_num_elem, beta_num_elem).CopyFromVec(Vector<BaseFloat>(beta_corr_));
+  }
+
+  void GetParams(VectorBase<BaseFloat>* params) const {
+    KALDI_ASSERT(params->Dim() == NumParams());
+    int32 alpha_num_elem = alpha_.Dim();
+    int32 beta_num_elem = beta_.Dim();
+    params->Range(0, alpha_num_elem).CopyFromVec(Vector<BaseFloat>(alpha_));
+    params->Range(alpha_num_elem, beta_num_elem).CopyFromVec(Vector<BaseFloat>(beta_));
+  }
+
+  void SetParams(const VectorBase<BaseFloat>& params) {
+    KALDI_ASSERT(params.Dim() == NumParams());
+    int32 alpha_num_elem = alpha_.Dim();
+    int32 beta_num_elem = beta_.Dim();
+    alpha_.CopyFromVec(params.Range(0, alpha_num_elem));
+    beta_.CopyFromVec(params.Range(alpha_num_elem, beta_num_elem));
+  }
+
+  std::string Info() const {
+    return std::string("\n  alpha") +
+      MomentStatistics(alpha_) +
+      ", bias-lr-coef " + ToString(bias_learn_rate_coef_) +
+      "\n  beta" + MomentStatistics(beta_) +
+      ", lr-coef " + ToString(learn_rate_coef_);
+  }
+  std::string InfoGradient() const {
+    return std::string("\n  alpha_grad") +
+      MomentStatistics(alpha_corr_) +
+      ", bias-lr-coef " + ToString(bias_learn_rate_coef_) +
+      "\n  beta_grad" + MomentStatistics(beta_corr_) +
+      ", lr-coef " + ToString(learn_rate_coef_);
+  }
+
+  void PropagateFnc(const CuMatrixBase<BaseFloat> &in,
+                    CuMatrixBase<BaseFloat> *out) {
+    // Multiply activations by ReLU scalars (max_alpha, min_beta):
+    // // out = in * (in >= 0.0 ? in * max_alpha : in * min_beta)
+    out->ParametricRelu(in, alpha_, beta_);
+  }
+
+  void BackpropagateFnc(const CuMatrixBase<BaseFloat> &in,
+                        const CuMatrixBase<BaseFloat> &out,
+                        const CuMatrixBase<BaseFloat> &out_diff,
+                        CuMatrixBase<BaseFloat> *in_diff) {
+    // multiply error derivative by activations (alpha, beta)
+    // ((dE/da)*w) === out_diff, f(y) == out,
+    //  (out > 0 ) ? out_diff * alpha : out_diff * beta
+    in_diff->DiffParametricRelu(out, out_diff, alpha_, beta_);
+  }
+
+
+  void Update(const CuMatrixBase<BaseFloat> &input,
+              const CuMatrixBase<BaseFloat> &diff) {
+    // we use following hyperparameters from the option class
+    const BaseFloat alr = opts_.learn_rate * bias_learn_rate_coef_;
+    const BaseFloat blr = opts_.learn_rate * learn_rate_coef_;
+    const BaseFloat mmt = opts_.momentum;
+    if (clip_gradient_ > 0.0) {  // gradient clipping
+      alpha_corr_.ApplyFloor(-clip_gradient_);
+      beta_corr_.ApplyFloor(-clip_gradient_);
+      alpha_corr_.ApplyCeiling(clip_gradient_);
+      beta_corr_.ApplyCeiling(clip_gradient_);
+    }
+    // compute gradient (incl. momentum)
+    if (!fix_alpha_) {  // the alpha parameter is learnable
```

Comment:
It would be better to check for 'a_lr > 0.0'. The 'fix_alpha_' and 'fix_beta_' can be removed.